### ISSUE TITLE
Using Spack on RockyLinux with GNU 12.2.0 results in a "C compiler cannot create executables" error

### ISSUE THREAD

**Issue by yantosca** [fasrc/User_Codes#10]
## Problem

On the RockyLinux test cluster, I tried to build a simple package (cgdb) with Spack as a test.  But the build fails because a configure script returns a "C compiler cannot create executables" error.  I'm doing this within an salloc job on holy7c12107 (with Intel CPUs).  I logged in directly to rockylogin.rc.fas.harvard.edu before starting the salloc job.

Maybe there is something I'm doing wrong... but I also have built Spack libraries for my own purposes before using similar steps.  But perhaps I've missed a setup step.  If you see anything please let me know.

## Error output

```console
==> Warning: Skipping build of libxml2-2.10.1-miqqfumjbisd327mcqs4yqcivi3vlech since zlib-1.2.13-xzya3i6ni4zkbycrk2bnbwba3dtfjpag failed
==> Warning: Skipping build of gettext-0.21.1-edshaontdafglijilmuyltki6jkbag7k since libxml2-2.10.1-miqqfumjbisd327mcqs4yqcivi3vlech failed
==> Warning: Skipping build of sqlite-3.39.4-na7itlovd25dqia2h3n5i3vktqtbt5lh since zlib-1.2.13-xzya3i6ni4zkbycrk2bnbwba3dtfjpag failed
==> Warning: Module file /net/holy-nfsisilon/ifs/rc_labs/jacob_lab/Lab/seasfs01/Lab/RockyLinux/spack/share/spack/modules/linux-rocky8-x86_64/flex-2.6.4-5dga27e exists and will not be overwritten
[+] /n/sw/helmod-rocky8/apps/Core/flex/2.6.4-fasrc01 (external flex-2.6.4-5dga27e2evtxg4grzdy2lijxss6ubs27)
[+] /usr (external autoconf-2.69-vwhhbbcnzj4zsknv6lr5p2754rbhba3u)
[+] /usr (external automake-1.16.1-3mtydvvas74ibdxyjibj2ju67hcrceuy)
[+] /usr (external libtool-2.4.6-m3rpviqsbbz5qzlwktkpkpky462xpl4r)
==> Installing libmd-1.0.4-jxvqhjl7uir2utgpdxzxho744wnbyicj
==> No binary for libmd-1.0.4-jxvqhjl7uir2utgpdxzxho744wnbyicj found: installing from source
==> Using cached archive: /net/holy-nfsisilon/ifs/rc_labs/jacob_lab/Lab/seasfs01/Lab/RockyLinux/spack/var/spack/cache/_source-cache/archive/f5/f51c921042e34beddeded4b75557656559cf5b1f2448033b4c1eec11c07e530f.tar.xz
==> No patches needed for libmd
==> libmd: Executing phase: 'autoreconf'
==> libmd: Executing phase: 'configure'
==> Error: ProcessError: Command exited with status 77:
    '/tmp/ryantosca/spack-stage/spack-stage-libmd-1.0.4-jxvqhjl7uir2utgpdxzxho744wnbyicj/spack-src/configure' '--prefix=/net/holy-nfsisilon/ifs/rc_labs/jacob_lab/Lab/seasfs01/Lab/RockyLinux/spack/opt/spack/linux-rocky8-x86_64/gcc-12.2.0/libmd-1.0.4-jxvqhjl7uir2utgpdxzxho744wnbyicj'

2 errors found in build log:
     11    checking whether make sets $(MAKE)... yes
     12    checking whether make supports nested variables... yes
     13    checking whether make supports nested variables... (cached) yes
     14    checking whether make supports the include directive... yes (GNU style)
     15    checking for gcc... /net/holy-nfsisilon/ifs/rc_labs/jacob_lab/Lab/seasfs01/Lab/RockyLinux/spack/lib/spack/env/
           gcc/gcc
     16    checking whether the C compiler works... no
  >> 17    configure: error: in `/tmp/ryantosca/spack-stage/spack-stage-libmd-1.0.4-jxvqhjl7uir2utgpdxzxho744wnbyicj/spac
           k-src':
  >> 18    configure: error: C compiler cannot create executables
     19    See `config.log' for more details

See build log for details:
  /tmp/ryantosca/spack-stage/spack-stage-libmd-1.0.4-jxvqhjl7uir2utgpdxzxho744wnbyicj/spack-build-out.txt
```

### spack-build-out.txt
```console
==> gmp: Executing phase: 'autoreconf'
==> [2023-04-20-17:00:00.038567] Configure script not found: trying to generate it
==> [2023-04-20-17:00:00.038653] Warning: *********************************************************
==> [2023-04-20-17:00:00.038681] Warning: * If the default procedure fails, consider implementing *
==> [2023-04-20-17:00:00.038700] Warning: *        a custom AUTORECONF phase in the package       *
==> [2023-04-20-17:00:00.038717] Warning: *********************************************************
==> [2023-04-20-17:00:00.039396] '/usr/bin/autoreconf' '-ivf'
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force
autoreconf: configure.ac: tracing
autoreconf: running: libtoolize --copy --force
libtoolize: putting auxiliary files in '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: Consider adding 'AC_CONFIG_MACRO_DIRS([m4])' to configure.ac,
libtoolize: and rerunning libtoolize and aclocal.
libtoolize: Consider adding '-I m4' to ACLOCAL_AMFLAGS in Makefile.am.
autoreconf: running: /usr/bin/autoconf --force
autoreconf: running: /usr/bin/autoheader --force
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:2472: installing './compile'
configure.ac:81: installing './missing'
autoreconf: Leaving directory `.'
==> gmp: Executing phase: 'configure'
==> [2023-04-20-17:00:05.989860] FILTER FILE: /tmp/ryantosca/spack-stage/spack-stage-gmp-6.2.1-rb23pkqphp45iv7izd3wdgq6tif
d4l3g/spack-src/configure [replacing "^(\s*if test x-L = )("\$p" \|\|\s*)$"]
==> [2023-04-20-17:00:06.028127] FILTER FILE: /tmp/ryantosca/spack-stage/spack-stage-gmp-6.2.1-rb23pkqphp45iv7izd3wdgq6tifd4l3g/spack-src/configure [replacing "^(\s*test x-R = )("\$p")(; then\s*)$"]
==> [2023-04-20-17:00:06.062429] FILTER FILE: /tmp/ryantosca/spack-stage/spack-stage-gmp-6.2.1-rb23pkqphp45iv7izd3wdgq6tifd4l3g/spack-src/configure [replacing "^(\s*test \$p = "-R")(; then\s*)$"]
==> [2023-04-20-17:00:06.099022] '/tmp/ryantosca/spack-stage/spack-stage-gmp-6.2.1-rb23pkqphp45iv7izd3wdgq6tifd4l3g/spack-src/configure' '--prefix=/net/holy-nfsisilon/ifs/rc_labs/jacob_lab/Lab/seasfs01/Lab/RockyLinux/spack/opt/spack/linux-rocky8-x86_64/gcc-12.2.0/gmp-6.2.1-rb23pkqphp45iv7izd3wdgq6tifd4l3g' '--enable-cxx' '--enable-shared' '--enable-static' '--with-pic'
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking ABI=64
checking compiler /net/holy-nfsisilon/ifs/rc_labs/jacob_lab/Lab/seasfs01/Lab/RockyLinux/spack/lib/spack/env/gcc/gcc -O2 -pedantic -fomit-frame-pointer -m64 ... no
checking ABI=x32
checking compiler /net/holy-nfsisilon/ifs/rc_labs/jacob_lab/Lab/seasfs01/Lab/RockyLinux/spack/lib/spack/env/gcc/gcc -O2 -pedantic -fomit-frame-pointer -mx32 ... no
checking ABI=32
checking compiler /net/holy-nfsisilon/ifs/rc_labs/jacob_lab/Lab/seasfs01/Lab/RockyLinux/spack/lib/spack/env/gcc/gcc -m32 -O2 -pedantic -fomit-frame-pointer ... no
checking compiler /net/holy-nfsisilon/ifs/rc_labs/jacob_lab/Lab/seasfs01/Lab/RockyLinux/spack/lib/spack/env/gcc/gcc -O2 -pedantic -fomit-frame-pointer ... no
configure: error: could not find a working compiler, see config.log for details
```
## Steps to reproduce
```console
# Navigate to target folder
cd /net/holy-nfsisilon/ifs/rc_labs/jacob_lab/Lab/seasfs01/Lab/RockyLinux

# Load modules
module purge
module load gcc/12.2.0-fasrc01
module load openmpi/4.1.4-fasrc01
module load netcdf-fortran/4.6.0-fasrc02
module load flex/2.6.4-fasrc01
module load cmake/3.25.2-fasrc01

# Setup spack
git clone -c feature.manyFiles=true https://github.com/spack/spack.git
cd spack
git checkout releases/v0.19
cd ..
 . spack/share/spack/setup-env.sh

# Look for externals and compilers
spack -C spack-env/gnu12 external find
spack -C spack-env/gnu12 compiler find

# Try building a simple package
spack -C spack-env/gnu12 install cgdb
```

### My /n/home09/ryantosca/.spack/linux/compilers.yaml file
```yaml
compilers:
- compiler:
    spec: gcc@12.2.0
    paths:
      cc: /n/sw/helmod-rocky8/apps/Core/gcc/12.2.0-fasrc01/bin/gcc
      cxx: /n/sw/helmod-rocky8/apps/Core/gcc/12.2.0-fasrc01/bin/g++
      f77: /n/sw/helmod-rocky8/apps/Core/gcc/12.2.0-fasrc01/bin/gfortran
      fc: /n/sw/helmod-rocky8/apps/Core/gcc/12.2.0-fasrc01/bin/gfortran
    flags: {}
    operating_system: rocky8
    target: x86_64
    modules: []
    environment: {}
    extra_rpaths: []
- compiler:
    spec: gcc@8.5.0
    paths:
      cc: /usr/bin/gcc
      cxx: /usr/bin/g++
      f77: /usr/bin/gfortran
      fc: /usr/bin/gfortran
    flags: {}
    operating_system: rocky8
    target: x86_64
    modules: []
    environment: {}
    extra_rpaths: []
```

### My /n/jacob_lab/Lab/seasfs01/Lab/RockyLinux/spack-env/gnu12/packages.yaml file
```yaml
packages:
  #
  # Package specs
  # 
  cgdb:
    version: [0.7.1]
    compiler: [gcc@12.2.0]
  #
  # Settings for all Spack-built packages
  #
  all:
    target: [x86_64]
```

**Comment by yantosca** [comment]
Hi @pedmon, thanks for the tip.  I figured it wasn't finding a library or something and that seemed to be the case.

So when I modified my `compilers.yaml` to include the library paths:
```yaml
- compiler:
    spec: gcc@12.2.0
    paths:
      cc: /n/sw/helmod-rocky8/apps/Core/gcc/12.2.0-fasrc01/bin/gcc
      cxx: /n/sw/helmod-rocky8/apps/Core/gcc/12.2.0-fasrc01/bin/g++
      f77: /n/sw/helmod-rocky8/apps/Core/gcc/12.2.0-fasrc01/bin/gfortran
      fc: /n/sw/helmod-rocky8/apps/Core/gcc/12.2.0-fasrc01/bin/gfortran
    flags: {}
    operating_system: rocky8
    target: x86_64
    modules: [gcc/12.2.0-fasrc01]
    environment:
      prepend_path:
        LIBRARY_PATH: /n/sw/helmod-rocky8/apps/Core/mpfr/4.2.0-fasrc01/lib64
        LD_LIBRARY_PATH: /n/sw/helmod-rocky8/apps/Core/mpfr/4.2.0-fasrc01/lib64
    extra_rpaths: []
```
then I no longer get the build error, and many of the dependencies are built:

```console
$ spack find
-- linux-rocky8-x86_64 / gcc@12.2.0 -----------------------------
autoconf@2.69                       diffutils@3.6   libiconv@1.16   openssl@1.1.1s  texinfo@6.5
automake@1.16.1                     flex@2.6.4      libmd@1.0.4     perl@5.36.0     util-linux-uuid@2.38.1
berkeley-db@18.1.40                 gdbm@1.23       libtool@2.4.6   pkgconf@1.4.2   xz@5.2.7
bison@3.8.2                         gettext@0.21.1  libxml2@2.10.1  readline@8.1.2  zlib@1.2.13
bzip2@1.0.8                         gmp@6.2.1       m4@1.4.18       sqlite@3.39.4
ca-certificates-mozilla@2022-10-11  libffi@3.4.2    ncurses@6.3     tar@1.30
==> 28 installed packages
```
But the cgdb build halts with an error.  I now get this error:

```console
==> Error: ProcessError: Command exited with status 2:
    'make' '-j8' 'V=1'


15 errors found in build log:
     314    mv -f $depbase.Tpo $depbase.Plo
     315    depbase=`echo pidfile.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
     316    /bin/sh ../libtool  --tag=CC   --mode=compile /net/holy-nfsisilon/ifs/rc_labs/jacob_lab/Lab/seasfs01/Lab/Ro
            ckyLinux/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H   -I.. -isystem ../include/bsd/ -include ../config.h -
            DLIBBSD_OVERLAY -DLIBBSD_DISABLE_DEPRECATED -D__REENTRANT    -g -O2  -Wall -Wextra -Wbad-function-cast -Wc9
            9-c11-compat -Wcast-align -Wdeclaration-after-statement -Wduplicated-branches -Wduplicated-cond -Wformat -W
            format-security -Wformat=2 -Winit-self -Wlogical-not-parentheses -Wlogical-op -Wmissing-declarations -Wmiss
            ing-format-attribute -Wmissing-prototypes -Wnested-externs -Wno-missing-field-initializers -Wno-nonnull-com
            pare -Wno-unused-parameter -Wnull-dereference -Wold-style-definition -Wpointer-arith -Wrestrict -Wshadow -W
            shift-negative-value -Wsizeof-array-argument -Wstrict-prototypes -Wswitch-bool -Wvla -Wwrite-strings -MT pi
            dfile.lo -MD -MP -MF $depbase.Tpo -c -o pidfile.lo pidfile.c &&\
     317    mv -f $depbase.Tpo $depbase.Plo
     318    libtool: compile:  /net/holy-nfsisilon/ifs/rc_labs/jacob_lab/Lab/seasfs01/Lab/RockyLinux/spack/lib/spack/en
            v/gcc/gcc -DHAVE_CONFIG_H -I.. -isystem ../include/bsd/ -include ../config.h -DLIBBSD_OVERLAY -DLIBBSD_DISA
            BLE_DEPRECATED -D__REENTRANT -g -O2 -Wall -Wextra -Wbad-function-cast -Wc99-c11-compat -Wcast-align -Wdecla
            ration-after-statement -Wduplicated-branches -Wduplicated-cond -Wformat -Wformat-security -Wformat=2 -Winit
            -self -Wlogical-not-parentheses -Wlogical-op -Wmissing-declarations -Wmissing-format-attribute -Wmissing-pr
            ototypes -Wnested-externs -Wno-missing-field-initializers -Wno-nonnull-compare -Wno-unused-parameter -Wnull
            -dereference -Wold-style-definition -Wpointer-arith -Wrestrict -Wshadow -Wshift-negative-value -Wsizeof-arr
            ay-argument -Wstrict-prototypes -Wswitch-bool -Wvla -Wwrite-strings -MT heapsort.lo -MD -MP -MF .deps/heaps
            ort.Tpo -c heapsort.c -o heapsort.o >/dev/null 2>&1
     319    libtool: compile:  /net/holy-nfsisilon/ifs/rc_labs/jacob_lab/Lab/seasfs01/Lab/RockyLinux/spack/lib/spack/en
            v/gcc/gcc -DHAVE_CONFIG_H -I.. -isystem ../include/bsd/ -include ../config.h -DLIBBSD_OVERLAY -DLIBBSD_DISA
            BLE_DEPRECATED -D__REENTRANT -g -O2 -Wall -Wextra -Wbad-function-cast -Wc99-c11-compat -Wcast-align -Wdecla
            ration-after-statement -Wduplicated-branches -Wduplicated-cond -Wformat -Wformat-security -Wformat=2 -Winit
            -self -Wlogical-not-parentheses -Wlogical-op -Wmissing-declarations -Wmissing-format-attribute -Wmissing-pr
            ototypes -Wnested-externs -Wno-missing-field-initializers -Wno-nonnull-compare -Wno-unused-parameter -Wnull
            -dereference -Wold-style-definition -Wpointer-arith -Wrestrict -Wshadow -Wshift-negative-value -Wsizeof-arr
            ay-argument -Wstrict-prototypes -Wswitch-bool -Wvla -Wwrite-strings -MT humanize_number.lo -MD -MP -MF .dep
            s/humanize_number.Tpo -c humanize_number.c -o humanize_number.o >/dev/null 2>&1
  >> 320    md5.c:33:16: error: unknown type name 'MD5_CTX'
     321       33 | libbsd_MD5Init(MD5_CTX *context)
     322          |                ^~~~~~~
     323    In file included from md5.c:28:
  >> 324    md5.c:40:29: error: 'libbsd_MD5Init' undeclared here (not in a function)
     325       40 | libbsd_symver_weak(MD5Init, libbsd_MD5Init, LIBBSD_0.0);
     326          |                             ^~~~~~~~~~~~~~
     327    local-link.h:47:27: note: in definition of macro 'libbsd_symver_weak'
     328       47 |         extern __typeof__(symbol) symbol \
     329          |                           ^~~~~~
  >> 330    md5.c:43:18: error: unknown type name 'MD5_CTX'
     331       43 | libbsd_MD5Update(MD5_CTX *context, const uint8_t *data, size_t len)
     332          |                  ^~~~~~~
  >> 333    md5.c:50:31: error: 'libbsd_MD5Update' undeclared here (not in a function); did you mean 'libbsd_MD5Init'?
     334       50 | libbsd_symver_weak(MD5Update, libbsd_MD5Update, LIBBSD_0.0);
     335          |                               ^~~~~~~~~~~~~~~~
     336    local-link.h:47:27: note: in definition of macro 'libbsd_symver_weak'
     337       47 |         extern __typeof__(symbol) symbol \
     338          |                           ^~~~~~
  >> 339    md5.c:53:15: error: unknown type name 'MD5_CTX'
     340       53 | libbsd_MD5Pad(MD5_CTX *context)
     341          |               ^~~~~~~
  >> 342    md5.c:60:28: error: 'libbsd_MD5Pad' undeclared here (not in a function); did you mean 'libbsd_MD5Init'?
     343       60 | libbsd_symver_weak(MD5Pad, libbsd_MD5Pad, LIBBSD_0.0);
     344          |                            ^~~~~~~~~~~~~
     345    local-link.h:47:27: note: in definition of macro 'libbsd_symver_weak'
     346       47 |         extern __typeof__(symbol) symbol \
     347          |                           ^~~~~~
  >> 348    md5.c:63:32: error: 'MD5_DIGEST_LENGTH' undeclared here (not in a function)
     349       63 | libbsd_MD5Final(uint8_t digest[MD5_DIGEST_LENGTH], MD5_CTX *context)
     350          |                                ^~~~~~~~~~~~~~~~~
  >> 351    md5.c:63:52: error: unknown type name 'MD5_CTX'
     352       63 | libbsd_MD5Final(uint8_t digest[MD5_DIGEST_LENGTH], MD5_CTX *context)
     353          |                                                    ^~~~~~~
  >> 354    md5.c:70:30: error: 'libbsd_MD5Final' undeclared here (not in a function); did you mean 'libbsd_MD5Init'?
     355       70 | libbsd_symver_weak(MD5Final, libbsd_MD5Final, LIBBSD_0.0);
     356          |                              ^~~~~~~~~~~~~~~
     357    local-link.h:47:27: note: in definition of macro 'libbsd_symver_weak'
     358       47 |         extern __typeof__(symbol) symbol \
     359          |                           ^~~~~~
  >> 360    md5.c:73:60: error: 'MD5_BLOCK_LENGTH' undeclared here (not in a function)
     361       73 | libbsd_MD5Transform(uint32_t state[4], const uint8_t block[MD5_BLOCK_LENGTH])
     362          |                                                            ^~~~~~~~~~~~~~~~
     363    md5.c: In function 'libbsd_MD5Transform':
     364    md5.c:75:9: warning: implicit declaration of function 'MD5Transform'; did you mean 'libbsd_MD5Transform'? [
            -Wimplicit-function-declaration]
     365       75 |         MD5Transform(state, block);
     366          |         ^~~~~~~~~~~~
     367          |         libbsd_MD5Transform
     368    md5.c:75:9: warning: nested extern declaration of 'MD5Transform' [-Wnested-externs]
     369    md5.c: At top level:
  >> 370    md5.c:83:15: error: unknown type name 'MD5_CTX'
     371       83 | libbsd_MD5End(MD5_CTX *context, char *buf)
     372          |               ^~~~~~~
  >> 373    md5.c:90:28: error: 'libbsd_MD5End' undeclared here (not in a function); did you mean 'libbsd_MD5Pad'?
     374       90 | libbsd_symver_weak(MD5End, libbsd_MD5End, LIBBSD_0.0);
     375          |                            ^~~~~~~~~~~~~
     376    local-link.h:47:27: note: in definition of macro 'libbsd_symver_weak'
     377       47 |         extern __typeof__(symbol) symbol \
     378          |                           ^~~~~~
     379    md5.c: In function 'libbsd_MD5File':

     ...

     398      115 |         return MD5Data(data, len, buf);
     399          |                ^~~~~~~
     400    md5.c:115:16: warning: nested extern declaration of 'MD5Data' [-Wnested-externs]
     401    md5.c:115:16: warning: returning 'int' from a function with return type 'char *' makes pointer from integer
             without a cast [-Wint-conversion]
     402      115 |         return MD5Data(data, len, buf);
     403          |                ^~~~~~~~~~~~~~~~~~~~~~~
  >> 404    make[2]: *** [Makefile:659: md5.lo] Error 1
     405    make[2]: *** Waiting for unfinished jobs....
     406    libtool: compile:  /net/holy-nfsisilon/ifs/rc_labs/jacob_lab/Lab/seasfs01/Lab/RockyLinux/spack/lib/spack/en
            v/gcc/gcc -DHAVE_CONFIG_H -I.. -isystem ../include/bsd/ -include ../config.h -DLIBBSD_OVERLAY -DLIBBSD_DISA
            BLE_DEPRECATED -D__REENTRANT -g -O2 -Wall -Wextra -Wbad-function-cast -Wc99-c11-compat -Wcast-align -Wdecla
            ration-after-statement -Wduplicated-branches -Wduplicated-cond -Wformat -Wformat-security -Wformat=2 -Winit
            -self -Wlogical-not-parentheses -Wlogical-op -Wmissing-declarations -Wmissing-format-attribute -Wmissing-pr
            ototypes -Wnested-externs -Wno-missing-field-initializers -Wno-nonnull-compare -Wno-unused-parameter -Wnull
            -dereference -Wold-style-definition -Wpointer-arith -Wrestrict -Wshadow -Wshift-negative-value -Wsizeof-arr
            ay-argument -Wstrict-prototypes -Wswitch-bool -Wvla -Wwrite-strings -MT pidfile.lo -MD -MP -MF .deps/pidfil
            e.Tpo -c pidfile.c  -fPIC -DPIC -o .libs/pidfile.o
     407    libtool: compile:  /net/holy-nfsisilon/ifs/rc_labs/jacob_lab/Lab/seasfs01/Lab/RockyLinux/spack/lib/spack/en
            v/gcc/gcc -DHAVE_CONFIG_H -I.. -isystem ../include/bsd/ -include ../config.h -DLIBBSD_OVERLAY -DLIBBSD_DISA
            BLE_DEPRECATED -D__REENTRANT -g -O2 -Wall -Wextra -Wbad-function-cast -Wc99-c11-compat -Wcast-align -Wdecla
            ration-after-statement -Wduplicated-branches -Wduplicated-cond -Wformat -Wformat-security -Wformat=2 -Winit
            -self -Wlogical-not-parentheses -Wlogical-op -Wmissing-declarations -Wmissing-format-attribute -Wmissing-pr
            ototypes -Wnested-externs -Wno-missing-field-initializers -Wno-nonnull-compare -Wno-unused-parameter -Wnull
            -dereference -Wold-style-definition -Wpointer-arith -Wrestrict -Wshadow -Wshift-negative-value -Wsizeof-arr
            ay-argument -Wstrict-prototypes -Wswitch-bool -Wvla -Wwrite-strings -MT nlist.lo -MD -MP -MF .deps/nlist.Tp
            o -c nlist.c  -fPIC -DPIC -o .libs/nlist.o
     408    libtool: compile:  /net/holy-nfsisilon/ifs/rc_labs/jacob_lab/Lab/seasfs01/Lab/RockyLinux/spack/lib/spack/en
            v/gcc/gcc -DHAVE_CONFIG_H -I.. -isystem ../include/bsd/ -include ../config.h -DLIBBSD_OVERLAY -DLIBBSD_DISA
            BLE_DEPRECATED -D__REENTRANT -g -O2 -Wall -Wextra -Wbad-function-cast -Wc99-c11-compat -Wcast-align -Wdecla
            ration-after-statement -Wduplicated-branches -Wduplicated-cond -Wformat -Wformat-security -Wformat=2 -Winit
            -self -Wlogical-not-parentheses -Wlogical-op -Wmissing-declarations -Wmissing-format-attribute -Wmissing-pr
            ototypes -Wnested-externs -Wno-missing-field-initializers -Wno-nonnull-compare -Wno-unused-parameter -Wnull
            -dereference -Wold-style-definition -Wpointer-arith -Wrestrict -Wshadow -Wshift-negative-value -Wsizeof-arr
            ay-argument -Wstrict-prototypes -Wswitch-bool -Wvla -Wwrite-strings -MT inet_net_pton.lo -MD -MP -MF .deps/
            inet_net_pton.Tpo -c inet_net_pton.c -o inet_net_pton.o >/dev/null 2>&1
     409    libtool: compile:  /net/holy-nfsisilon/ifs/rc_labs/jacob_lab/Lab/seasfs01/Lab/RockyLinux/spack/lib/spack/en
            v/gcc/gcc -DHAVE_CONFIG_H -I.. -isystem ../include/bsd/ -include ../config.h -DLIBBSD_OVERLAY -DLIBBSD_DISA
            BLE_DEPRECATED -D__REENTRANT -g -O2 -Wall -Wextra -Wbad-function-cast -Wc99-c11-compat -Wcast-align -Wdecla
            ration-after-statement -Wduplicated-branches -Wduplicated-cond -Wformat -Wformat-security -Wformat=2 -Winit
            -self -Wlogical-not-parentheses -Wlogical-op -Wmissing-declarations -Wmissing-format-attribute -Wmissing-pr
            ototypes -Wnested-externs -Wno-missing-field-initializers -Wno-nonnull-compare -Wno-unused-parameter -Wnull
            -dereference -Wold-style-definition -Wpointer-arith -Wrestrict -Wshadow -Wshift-negative-value -Wsizeof-arr
            ay-argument -Wstrict-prototypes -Wswitch-bool -Wvla -Wwrite-strings -MT pidfile.lo -MD -MP -MF .deps/pidfil
            e.Tpo -c pidfile.c -o pidfile.o >/dev/null 2>&1
     410    libtool: compile:  /net/holy-nfsisilon/ifs/rc_labs/jacob_lab/Lab/seasfs01/Lab/RockyLinux/spack/lib/spack/en
            v/gcc/gcc -DHAVE_CONFIG_H -I.. -isystem ../include/bsd/ -include ../config.h -DLIBBSD_OVERLAY -DLIBBSD_DISA
            BLE_DEPRECATED -D__REENTRANT -g -O2 -Wall -Wextra -Wbad-function-cast -Wc99-c11-compat -Wcast-align -Wdecla
            ration-after-statement -Wduplicated-branches -Wduplicated-cond -Wformat -Wformat-security -Wformat=2 -Winit
            -self -Wlogical-not-parentheses -Wlogical-op -Wmissing-declarations -Wmissing-format-attribute -Wmissing-pr
            ototypes -Wnested-externs -Wno-missing-field-initializers -Wno-nonnull-compare -Wno-unused-parameter -Wnull
            -dereference -Wold-style-definition -Wpointer-arith -Wrestrict -Wshadow -Wshift-negative-value -Wsizeof-arr
            ay-argument -Wstrict-prototypes -Wswitch-bool -Wvla -Wwrite-strings -MT nlist.lo -MD -MP -MF .deps/nlist.Tp
            o -c nlist.c -o nlist.o >/dev/null 2>&1
     411    libtool: compile:  /net/holy-nfsisilon/ifs/rc_labs/jacob_lab/Lab/seasfs01/Lab/RockyLinux/spack/lib/spack/en
            v/gcc/gcc -DHAVE_CONFIG_H -I.. -isystem ../include/bsd/ -include ../config.h -DLIBBSD_OVERLAY -DLIBBSD_DISA
            BLE_DEPRECATED -D__REENTRANT -g -O2 -Wall -Wextra -Wbad-function-cast -Wc99-c11-compat -Wcast-align -Wdecla
            ration-after-statement -Wduplicated-branches -Wduplicated-cond -Wformat -Wformat-security -Wformat=2 -Winit
            -self -Wlogical-not-parentheses -Wlogical-op -Wmissing-declarations -Wmissing-format-attribute -Wmissing-pr
            ototypes -Wnested-externs -Wno-missing-field-initializers -Wno-nonnull-compare -Wno-unused-parameter -Wnull
            -dereference -Wold-style-definition -Wpointer-arith -Wrestrict -Wshadow -Wshift-negative-value -Wsizeof-arr
            ay-argument -Wstrict-prototypes -Wswitch-bool -Wvla -Wwrite-strings -MT merge.lo -MD -MP -MF .deps/merge.Tp
            o -c merge.c -o merge.o >/dev/null 2>&1
     412    make[2]: Leaving directory '/tmp/ryantosca/spack-stage/spack-stage-libbsd-0.11.5-wem5x2oesmhxmfp7x63sgdop57
            ribigd/spack-src/src'
  >> 413    make[1]: *** [Makefile:412: all-recursive] Error 1
     414    make[1]: Leaving directory '/tmp/ryantosca/spack-stage/spack-stage-libbsd-0.11.5-wem5x2oesmhxmfp7x63sgdop57
            ribigd/spack-src'
  >> 415    make: *** [Makefile:344: all] Error 2

See build log for details:
  /tmp/ryantosca/spack-stage/spack-stage-libbsd-0.11.5-wem5x2oesmhxmfp7x63sgdop57ribigd/spack-build-out.txt
```
This looks like a C-language issue deep inside the code.
```console
  >> 333    md5.c:50:31: error: 'libbsd_MD5Update' undeclared here (not in a function); did you mean 'libbsd_MD5Init'?
     334       50 | libbsd_symver_weak(MD5Update, libbsd_MD5Update, LIBBSD_0.0);
     335          |                               ^~~~~~~~~~~~~~~~
```
I can try adding some compiler flags to the build for it to see if I get past the error.  I don't really need the cgdb package but thought that was a good test case.  It makes gdb a little easier to use but isn't crucial.

I will be doing some Spack builds with the GNU 10.2.0 and GNU 12.2.0 (things like NCO and CDO,, etc) so I'll keep you posted if I run into other issues.  At least the compilers are now able to make executables with your fix.

**Comment by pedmon** [comment]
Yeah, as you are undoubtedly aware between gcc versions they can change the default C standard they use. This can then break builds that were formerly working. I'm wondering if cgdb is not happy with the defaults in gcc 12.

Since this issue is resolved, I will resolve this specifically and then if you hit another issue please either file a ticket or an issue here.

**Comment by yantosca** [comment]
@pedmon: PostScript: I was able to build cgdb by setting this in compilers.yaml:
```yaml
    flags:
      cflags: -std=c99
```
This was trial and error.  I started with `-ansi` and that didn't work.  But the newer C99 does.  Like you said, the source code doesn't like the newer C language standards.
